### ISSUE TITLE
`(get, default)` vs. OpAssignOp

### ISSUE THREAD

**Issue by Simn** [HaxeFoundation/haxe#9750]
```haxe
class PropertyClass {
	public var propGet(get, default) = 0;

	public function new() {}

	function get_propGet() {
		return propGet;
	}
}

function main() {
	var c = new PropertyClass();
	c.propGet += 1;
	trace(c.propGet);
}
```

Generated JavaScript:

```js
function Main_main() {
	var c = new PropertyClass();
	c.propGet += 1;
	console.log("source/Main.hx:14:",c.get_propGet());
}
```

This ignores the getter. I believe it should instead generate `c.propGet = c.get_propGet() + 1`.

Any opinions?

**Comment by Simn** [comment]
Also, postfix unops should then generate this:

```haxe
	{
		var tmp = c.get_propGet();
		c.propGet = tmp + 1;
		tmp;
	}
```

Prefix unops are the same as `OpAssignOp`.

**Comment by haxiomic** [comment]
I think I've relied this working in past projects (which must be subtly broken)
I can't see any argument against `c.get_propGet() + 1`, where I'd assume `x += y` is equivalent to `x = x + y`

**Comment by RealyUniqueName** [comment]
> I think I've relied this working in past projects 

`(get,default)` combination was implemented only in 4.1

**Comment by Simn** [comment]
> (get,default) combination was implemented only in 4.1

Ah, I wasn't aware of that. In that case it wasn't implemented very well. :P

**Comment by haxiomic** [comment]
@RealyUniqueName You're right, I'm being dumb, I was thinking about `(get, set)` which I assumed could behave the same with `+=` but sure enough it works correctly
